src/python/CMSSpark/reports/aggregate_campaigns.py

```diff
@@ -137,22 +137,23 @@ def run(fout, date, yarn=None, verbose=None, patterns=None, antipatterns=None, i
 
     extract_campaign_udf = udf(lambda dataset: dataset.split('/')[2])
 
-    # dataset, size, dataset_access_type_id
-    dbs_df = fdf_df.join(ddf_df, fdf_df.f_dataset_id == ddf_df.d_dataset_id)\
-                   .drop('f_dataset_id')\
-                   .drop('d_dataset_id')\
-                   .withColumnRenamed('d_dataset', 'dataset')\
-                   .withColumnRenamed('f_file_size', 'size')\
+    # d_dataset_id, d_dataset, dataset_access_type
+    dbs_df = ddf_df.join(daf_df, ddf_df.d_dataset_access_type_id == daf_df.dataset_access_type_id)\
+                   .drop(ddf_df.d_dataset_access_type_id)\
```

Comment:
Could you please check that .drop here does not do any harm. My worry is that you just joined two tables and immediately drop the foreign key you're joining. I would suggest dropping columns at the end when you construct all joins.

Reply:
I just checked and it appears that it makes no difference. I tried dropping columns in the new expression after join is completed and the number of campaigns is the same.

Reply:
ok, I thought it was worth of checking anyhow. Let me know when you'll be done with the rest of comments.

Reply:
Ok, I'm now working on few more attempts to solve the leftover campaigns issue. I'll report my findings shortly.